daemon/loop.go

```diff
@@ -199,6 +199,52 @@ func (d *Daemon) doSync(logger log.Logger, lastKnownSyncTagRev *string, warnedAb
 		return err
 	}
 
+	// Retrieve commits we are going to work with. We do this early as a signature validation
+	// error may occur, resulting in a change to the newTagRev.
+	var initialSync bool
+	var commits []git.Commit
+	{
```

Comment:
Sorry if this is uncalled for (the PR is still a draft) but I think it would be way cleaner if we factored this code out of `doSync()` which is already humongous in size.

Reply:
Oh no, early reviews are more than welcome and I fully agree the size of `doSync()` is outrageous.

The reason I did (not yet) factored it out is that this part of Flux is relatively new to me and I would like to have the logic (and order of it) working first before deciding how we should shrink the size.